File: next-main/src/main/java/com/next/reservations/api/ReservationTimeController.java
package com.next.reservations.api;

import com.next.reservations.web.mapper.ReservationTimeMapper;
import com.next.reservations.web.request.ReservationTimeConfigurationRequest;
import com.next.reservations.web.request.ReservationTimeRequest;
import com.next.reservations.web.response.ReservationTimeResponse;
import com.next.shared.domain.Option;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("api/reservation-times")
public class ReservationTimeController {

    private ReservationTimeMapper mapper;

    public ReservationTimeController(ReservationTimeMapper mapper) {
        this.mapper = mapper;
    }

    @PostMapping
    public Long createOrUpdateTime(@RequestBody ReservationTimeRequest reservationTimeRequest) {
        return mapper.createOrUpdateReservationTime(reservationTimeRequest);
    }

    @GetMapping("{configId}/by-config")
    public List<ReservationTimeResponse> findAllReservationTimes(@PathVariable Long configId){
        return mapper.findAllReservationTimesByConfigId(configId);
    }

    @PostMapping("/config")
    public long createOrUpdateConfig(@RequestBody ReservationTimeConfigurationRequest reservationTimeConfigurationRequest){
        return mapper.createOrUpdateConfiguration(reservationTimeConfigurationRequest);
    }

    @GetMapping("by-date/{date}")
    public List<Option> findAllByDate(@PathVariable String date){
        return mapper.getTimesForDate(date);
    }
}

File: next-main/src/main/kotlin/com/next/reservations/api/ReservationTimeController.kt
package com.next.reservations.api

import com.next.reservations.web.mapper.ReservationTimeMapper
import com.next.reservations.web.request.ReservationTimeRequest
import com.next.reservations.web.response.ReservationTimeResponse
import com.next.shared.domain.Option
import org.springframework.web.bind.annotation.*


@RestController
@RequestMapping("api/reservation-times")
class ReservationTimeController(private val mapper: ReservationTimeMapper) {

    @PostMapping
    fun createOrUpdateTime(@RequestBody reservationTimeRequest: ReservationTimeRequest): Long {
        return mapper.createOrUpdateReservationTime(reservationTimeRequest)
    }

    @GetMapping("{configId}/by-config")
    fun findAllReservationTimesByConfig(@PathVariable configId: Long): List<ReservationTimeResponse> {
        return mapper.findAllReservationTimesByConfigId(configId)
    }

    @GetMapping("by-date/{date}")
    fun findAllByDate(@PathVariable date: String): List<Option> {
        return mapper.getTimesForDate(date)
    }

    @DeleteMapping("/{id}")
    fun deleteById(@PathVariable id: Long) {
        return mapper.deleteById(id)
    }
}
